packages/autocomplete-core/src/__tests__/getSources.test.ts

```diff
@@ -37,7 +41,98 @@ describe('getSources', () => {
     });
   });
 
-  test.todo('provides default source implementations');
+  test('provides a default source implementations', async () => {
+    const onStateChange = jest.fn();
+    const sources = createSource();
+    const getSources = () => {
+      return [
+        {
+          ...sources,
+          templates: {
+            item() {},
+          },
+        },
+      ];
+    };
+    const { inputElement } = createPlayground(createAutocomplete, {
+      getSources,
+      onStateChange,
+    });
+
+    inputElement.focus();
+    userEvent.type(inputElement, 'a');
+
+    await runAllMicroTasks();
+
+    expect(onStateChange).toHaveBeenCalledWith({
+      prevState: expect.anything(),
+      state: expect.objectContaining({
+        collections: expect.arrayContaining([
+          expect.objectContaining({
+            source: expect.objectContaining({
+              getItemInputValue: sources.getItemInputValue,
+              getItemUrl: sources.getItemUrl,
+              getItems: sources.getItems,
```

Comment:
That's enough for the test:

```suggestion
              getItemInputValue: expect.any(Function),
              getItemUrl: expect.any(Function),
              getItems: expect.any(Function),
```